loopy/target/opencl.py

```diff
@@ -468,31 +434,25 @@ def opencl_preamble_generator(preamble_info):
                 """ % dict(idx_ctype=kernel.target.dtype_to_typename(
                     kernel.index_dtype))))
 
-    for func in preamble_info.seen_functions:
-        if func.name == "pow" and func.c_name == "powf32":
-            yield("08_clpowf32", """
-            inline float powf32(float x, float y) {
-              return pow(x, y);
-            }""")
-
-        if func.name == "pow" and func.c_name == "powf64":
-            yield("08_clpowf64", """
-            inline double powf64(double x, double y) {
-              return pow(x, y);
-            }""")
-
 # }}}
 
 
 # {{{ expression mapper
 
 class ExpressionToOpenCLCExpressionMapper(ExpressionToCExpressionMapper):
-
     def wrap_in_typecast_lazy(self, actual_dtype, needed_dtype, s):
-        if needed_dtype.dtype.kind == "b" and actual_dtype().dtype.kind == "f":
+        _actual = actual_dtype()
```

Comment:
I think the idea behind `wrap_in_typecast_lazy` was to only do type inference (which is what's behind the `actual_dtype` callable) if it is clear from the circumstances that it is needed (and avoid in the majority of cases). Since this calls type inference unconditionally, I think `wrap_in_typecast_lazy` has outlived its usefulness (It's only used in three spots any more), and we should probably just get rid of it in favor of always using `wrap_in_typecast`.

@kaushikcfd, do you agree?

Reply:
Yup, I don't see any issues in axing the lazy version.

Reply:
Cool, I can take care of it here unless you object.

Reply:
Done. I don't think it introduced any additional failures - sorry it won't be easy to tell from the CI runs. (Figured it'd be easier to just remove it now before debugging the rest.)